### PR TITLE
Improv/ctrl click to open in new tab

### PR DESCRIPTION
**Pull request by iannbing**
## Description

User might want to use Mouse middle click, or Ctrl + click (Cmd + click on Mac) to open a new tab if possible. 

## Checklist

- [x] The PR branch is set up to merge into `dev` instead of `master`.

### CODE REVIEW

File: services/orchest-webserver/client/src/hooks/useCustomRoute.ts [hunk 1]
```diff
@@ -107,16 +108,28 @@ const useCustomRoute = () => {
...
       const { query = null, state = {} } = params || {};
-      history.push({
-        pathname,
-        search: existingQueryString
-          ? `${existingQueryString}&${toQueryString(query)}`
-          : toQueryString(query),
-        state,
-      });
+
+      const shouldOpenNewTab = e?.ctrlKey || e?.metaKey;
```
Comment (yannickperrenet) [inline services/orchest-webserver/client/src/hooks/useCustomRoute.ts:119]: `ctrl` + click works like a charm for me. However, pressing the scroll wheel on my mouse doesn't do anything.

**Comment by ricklamers** [comment]
Great work @iannbing. Bar middle mouse click it looks good to me 👌🏻

File: services/orchest-webserver/client/src/utils/openInNewTab.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const openInNewTab = (url: string) => {
+  if (typeof window === undefined) return;
+
+  const link = document.createElement("a");
+  link.href = url;
+  link.target = "_blank";
+  document.body.appendChild(link);
+  link.click();
+  link.remove();
```
Comment (yannickperrenet) [inline services/orchest-webserver/client/src/utils/openInNewTab.ts:9]: Funny, didn't know it works like that ^_^

Comment (iannbing) [reply services/orchest-webserver/client/src/utils/openInNewTab.ts:9]: It's a necessary hack to bypass pop-up blocker :-)

**Review by yannickperrenet — APPROVED** [review]
Works like a charm for me now :sunglasses:

**Pull request merged**